File: Find_Odd_Number.cs
using System;

class Program
{
    // Method to check if a string represents a valid number
    static bool IsNumber(string str)
    {
        if (string.IsNullOrWhiteSpace(str))
        {
            return false;
        }

        foreach (char c in str)
        {
            if (!char.IsDigit(c))
            {
                return false;
            }
        }
        return true;
    }

    // Method to print odd numbers between two values
    static void PrintOddNumbersBetween(string input1, string input2)
    {
        if (int.TryParse(input1, out int m) && int.TryParse(input2, out int n))
        {
            // Ensure m is the smaller number and n is the larger number
            int start = Math.Min(m, n);
            int end = Math.Max(m, n);

            Console.WriteLine($"The Odd Numbers Between {start} and {end}:");

            // Print odd numbers between start and end
            for (int i = start; i <= end; i++)
            {
                if (i % 2 != 0)
                {
                    Console.Write(i + " ");
                }
            }
            Console.WriteLine();
        }
        else
        {
            Console.WriteLine("Enter valid numbers as values!");
        }
    }

    static void Main()
    {
        Console.WriteLine("Enter two values:");
        string input1 = Console.ReadLine();
        string input2 = Console.ReadLine();

        PrintOddNumbersBetween(input1, input2);
    }
}

File: Find_Odd_Number.cpp
#include <iostream>
#include <string>
#include <cctype>
#include <algorithm>
#include <stdexcept>

// Function to check if a string represents a valid number
bool isNumber(const std::string& str) {
    if (str.empty()) {
        return false;
    }
    return std::all_of(str.begin(), str.end(), ::isdigit);
}

// Function to print odd numbers between two values
void printOddNumbersBetween(const std::string& input1, const std::string& input2) {
    try {
        if (isNumber(input1) && isNumber(input2)) {
            int m = std::stoi(input1); // Convert string to integer
            int n = std::stoi(input2);

            // Ensure m is the smaller number and n is the larger number
            int start = std::min(m, n);
            int end = std::max(m, n);

            std::cout << "The Odd Numbers Between " << start << " and " << end << ":" << std::endl;

            // Print odd numbers between start and end
            for (int i = start; i <= end; ++i) {
                if (i % 2 != 0) {
                    std::cout << i << " ";
                }
            }
            std::cout << std::endl;
        } else {
            std::cout << "Enter valid numbers as values!" << std::endl;
        }
    } catch (const std::invalid_argument& e) {
        std::cout << "Error: Invalid number format." << std::endl;
    } catch (const std::out_of_range& e) {
        std::cout << "Error: Number out of range." << std::endl;
    }
}

int main() {
    std::string input1, input2;

    std::cout << "Enter two values: " << std::endl;
    std::cin >> input1 >> input2;

    printOddNumbersBetween(input1, input2);

    return 0;
}
